Sync/SyncTelemetryUtils.swift

```diff
@@ -384,3 +384,72 @@ public struct SyncPing: SyncTelemetryPing {
         }
     }
 }
+
+public class GleanSyncOperationHelper {
+    public init () {}
+
+    public func start() {
+        GleanMetrics.Sync.syncUuid.generateAndSet()
+    }
+
+    public func end(_ result: SyncOperationResult) {
+        if let engineResults = result.engineResults.successValue {
+            engineResults.forEach { result in
+                let (name, status) = result
+                switch status {
+                case .completed(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .partial(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .notStarted(let reason):
+                    ()
+                    // failure_reason would be recorded here for the temp-<engine>-sync pings
+                }
+            }
+        } else if let failure = result.engineResults.failureValue {
+            // failure_reason would be recorded here for the temp-sync ping
+        }
+        
+        GleanMetrics.Pings.tempSync.submit()
+    }
+    
+    private func submitPingPerEngine(_ engineName: String, _ stats: SyncEngineStatsSession) {
+        // Create maps on labels to stat value,
+        // keeping only the values that are above zero.
+        //
+        // If we attempt to add 0 to a Glean counter,
+        // Glean will record an error. We don't want that here.
+        let incomingLabelsToValue = [
+            // TODO: What is the difference between applied and succeeded?
+            ("applied", stats.downloadStats.applied + stats.downloadStats.succeeded),
+            ("reconciled", stats.downloadStats.reconciled),
+            // TODO: What is the difference between failed and newFailed?
+            ("failed_to_apply", stats.downloadStats.failed + stats.downloadStats.newFailed)
+        ].filter { (_, stat) in stat > 0 }
+        let outgoingLabelsToValue = [
+            ("uploaded", stats.uploadStats.sent),
+            ("failed_to_upload", stats.uploadStats.sentFailed)
+        ].filter { (_, stat) in stat > 0 }
+        
+        switch engineName {
```

Comment:
I am assuming creditcards and addresses are not synced in iOS. Am I assuming correctly?

Reply:
Correct. Credit cards and addresses are not currently synced.